pkg/sentry/fsimpl/erofs/erofs.go

```diff
@@ -243,6 +262,14 @@ type inode struct {
 	// +checklocks:dirMu
 	dirents []vfs.Dirent `state:"nosave"`
 
+	// mapsMu protects mappings.
+	mapsMu sync.Mutex `state:"nosave"`
+
+	// mappings tracks the mappings of the file into memmap.MappingSpaces
+	// if this inode represents a regular file.
+	// +checklocks:mapsMu
+	mappings memmap.MappingSet
+
```

Comment:
They are needed for C/R, because only the pmas for which `file == MemoryManager.mfp.MemoryFile()` may be saved:

https://github.com/google/gvisor/blob/37edf7285f2ab09381eba0203d7e14dfd85a4626/pkg/sentry/mm/mm.go#L438-L442

https://github.com/google/gvisor/blob/37edf7285f2ab09381eba0203d7e14dfd85a4626/pkg/sentry/mm/save_restore.go#L39-L45

and we need to track the mappings and invalidate these unsavable mappings in `memmap.Mappable.InvalidateUnsavable`:

https://github.com/google/gvisor/blob/37edf7285f2ab09381eba0203d7e14dfd85a4626/pkg/sentry/mm/save_restore.go#L25-L30

https://github.com/google/gvisor/blob/95397d22973675bb235c0819b36b067d85d7ea30/pkg/sentry/fsimpl/erofs/regular_file.go#L183-L188

Otherwise checkpoint will fail like this:

```
checkpoint failed: checkpointing container "test-container-seed": encoding error at object mm.MemoryManager{p:(*kernel.Kernel)(0xc000382a80), ...... }: Can't save pma hostarch.AddrRange{Start:0x563d21595000, End:0x563d21598000} with non-MemoryFile of type *erofs.imageMemmapFile:
......
PMAs:                                                                                                                                                                        
563d21593000-563d21595000 r--p 00003000 *pgalloc.MemoryFile
563d21595000-563d21598000 r-xs 00fc0000 *erofs.imageMemmapFile
563d21598000-563d21599000 r-xp 00005000 *pgalloc.MemoryFile
......
```